pkg/controller/render/render_controller.go

```diff
@@ -630,3 +722,522 @@ func getMachineConfigsForPool(pool *mcfgv1.MachineConfigPool, configs []*mcfgv1.
+	for _, imageStreamName := range ensureImageStreams {
+
+		// Check to see if we have the imagestream already
+		_, err = ctrl.imageclient.ImageV1().ImageStreams(ctrlcommon.MCONamespace).Get(context.TODO(), imageStreamName, metav1.GetOptions{})
+		if apierrors.IsNotFound(err) {
+
+			// It doesn't exist, so we need to make it, otherwise our builds will fail
+			newImageStream := &imagev1.ImageStream{}
+			newImageStream.Name = imageStreamName
+			newImageStream.Namespace = ctrlcommon.MCONamespace
+			newImageStream.Spec.LookupPolicy.Local = false
+
+			// Set ownerships so these get cleaned up if we delete the pool
+			// TODO(jkyros): I have no idea if this actually cleans the images out of the stream if we delete it?
+			poolKind := mcfgv1.SchemeGroupVersion.WithKind("MachineConfigPool")
+			oref := metav1.NewControllerRef(pool, poolKind)
+			newImageStream.SetOwnerReferences([]metav1.OwnerReference{*oref})
+
+			// coreos imagestream is base, it's special, it needs to pull that image
+			if imageStreamName == "coreos" {
+
+				// This gets the "rhel-coreos" image from our release out of our controller config
+				controllerConfig, err := ctrl.ccLister.Get(ctrlcommon.ControllerConfigName)
+				if err != nil {
+					return fmt.Errorf("could not get ControllerConfig %w", err)
+				}
+
+				newImageStream.Spec = imagev1.ImageStreamSpec{
+					LookupPolicy:          imagev1.ImageLookupPolicy{Local: false},
+					DockerImageRepository: "",
+					Tags: []imagev1.TagReference{
+						{
+							Name: "latest",
+							From: &corev1.ObjectReference{
+								Kind: "DockerImage",
+								Name: controllerConfig.Spec.BaseOperatingSystemContainer,
+							},
+						},
+					},
+				}
+
+			}
+
+			// It didn't exist, put the imagestream in the cluster
+			_, err := ctrl.imageclient.ImageV1().ImageStreams(ctrlcommon.MCONamespace).Create(context.TODO(), newImageStream, metav1.CreateOptions{})
+			if err != nil {
+				return fmt.Errorf("Attempted to create ImageStream %s but failed: %w", newImageStream, err)
+			}
+			glog.Infof("Created image stream %s", imageStreamName)
+		} else if err != nil {
+			// I don't know if it existed or not, I couldn't get it
+			return fmt.Errorf("Failed to retrieve ImageStream %s: %w", imageStreamName, err)
+		}
+
+		glog.Infof("ImageStream %s already exists", imageStreamName)
+	}
+
+	return nil
+}
+
+// experimentalAddBuildConfigs creates a build config for a pool so that its content automatically
+// gets rendered into an image when the machineconfig content changes
+func (ctrl *Controller) experimentalAddBuildConfigs(pool *mcfgv1.MachineConfigPool) error {
+
+	buildConfigName := fmt.Sprintf("mco-build-content-%s", pool.Name)
+	contentImageStreamName := pool.Name + ctrlcommon.ImageStreamSuffixMCOContent
+
+	// Multistage build, butane is optional (the [l] character class makes it a 'wildcard' and it won't fail if it doesn't exist)
+	var dockerFile = `
```

Comment:
If we want repeatable images, the dockerfile has to go away at some point (because otherwise we'd have to capture versions of it as part of the config or something) but I don't know that we are ready with a better option for now.